packages/devtools_app/lib/src/screens/performance/performance_controller.dart

```diff
@@ -124,11 +125,19 @@ class PerformanceController extends DevToolsScreenController
 
       if (serviceConnection.serviceManager.connectedApp?.isFlutterAppNow ??
           false) {
-        final impellerEnabledResponse = await serviceConnection.serviceManager
-            .callServiceExtensionOnMainIsolate(registrations.isImpellerEnabled);
-        _impellerEnabled = impellerEnabledResponse.json?['enabled'] == true;
+        // Do not await this future because this will hang if the app is paused
+        // upon connection.
+        unawaited(
+          serviceConnection.serviceManager
+              .callServiceExtensionOnMainIsolate(
```

Comment:
It might be worth auditing the locations where we invoke service extensions to make sure we're not awaiting responses anywhere that could possibly cause a hang.

Reply:
filed https://github.com/flutter/devtools/issues/9164 to track